### PR TITLE
Blogs-1327 OpenGraph Tags

### PR DESCRIPTION
**Pull request by mattgreenham**

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: templates/author/index.html.twig [hunk 1]
```diff
@@ -2,6 +2,18 @@
 
 {% block title %}BBC {{ tr('page_title_blogs') }} - {{ blog.getName() }} - {{ tr('page_title_author_all') }}{% endblock %}
 
+{% block social_tags %}
+    <meta property="og:type" content="website">
+    <meta property="og:site_name" content="{{ blog.getName() }}" />
```
Comment (votemike) [inline templates/author/index.html.twig]: Should `og:site_name` not be BBC for every page? http://ogp.me/#optional

File: templates/tag/index.html.twig [hunk 2]
```diff
@@ -2,6 +2,18 @@
 
 {% block title %}BBC {{ tr('page_title_blogs') }} - {{ blog.getName() }} - {{ tr('page_title_tag_list') }}{% endblock %}
 
+{% block social_tags %}
+    <meta property="og:type" content="website">
```
Comment (votemike) [inline templates/tag/index.html.twig]: This seems to be repeated everywhere, can we just put it in the base template?

File: templates/author/index.html.twig [hunk 3]
```diff
@@ -2,6 +2,18 @@
...
+    <meta property="og:title" content="{{ block('title') }}">
+    {%- if blog.getImage() -%}
+        <meta itemprop="image" content="{{ blog.getImage().getUrl(1200,675) }}">
+        <meta property="og:image" content="{{ blog.getImage().getUrl(1200,675) }}" />
+    {%- endif -%}
+    <meta property="og:description" content="{{ blog.getDescription() }}" />
```
Comment (votemike) [inline templates/author/index.html.twig]: Do we want the blog description as the description for the Authors Index page?

File: templates/author/index.html.twig [hunk 4]
```diff
@@ -2,6 +2,18 @@
...
+    <meta property="og:url" content="{{ url('author_index', {blogId: blog.getId()}) }}" />
```
Comment (votemike) [inline templates/author/index.html.twig]: Can this be genericised and put in the base template?

File: templates/author/show.html.twig [hunk 5]
```diff
@@ -2,6 +2,21 @@
...
+    <meta property="og:site_name" content="{{ blog.getName() }}" />
+    <meta property="og:title" content="{{ block('title') }}">
+    {%- if author.getImage() -%}
+        <meta itemprop="image" content="{{ author.getImage().getUrl(1200,675) }}">
+        <meta property="og:image" content="{{ author.getImage().getUrl(1200,675) }}" />
+    {%- elseif blog.getImage() -%}
+        <meta itemprop="image" content="{{ blog.getImage().getUrl(1200,675) }}">
+        <meta property="og:image" content="{{ blog.getImage().getUrl(1200,675) }}" />
+    {%- endif -%}
+    <meta property="og:description" content="{{ blog.getDescription() }}" />
```
Comment (votemike) [inline templates/author/show.html.twig]: Not convinced this would be the best description.....

File: templates/author/index.html.twig (hunk 4 above)
Comment (votemike) [reply templates/author/index.html.twig]: In programmes-frontend, there is also a `<link rel="canonical"` tag. Maybe add that in too?

File: templates/author/index.html.twig [hunk 6]
```diff
@@ -2,6 +2,18 @@
 
 {% block title %}BBC {{ tr('page_title_blogs') }} - {{ blog.getName() }} - {{ tr('page_title_author_all') }}{% endblock %}
 
+{% block social_tags %}
+    <meta property="og:type" content="website">
+    <meta property="og:site_name" content="{{ blog.getName() }}" />
+    <meta property="og:title" content="{{ block('title') }}">
+    {%- if blog.getImage() -%}
```
Comment (votemike) [inline templates/author/index.html.twig]: Tempted to make a new block in the blogs base template for this...

File: templates/author/show.html.twig [hunk 7]
```diff
@@ -2,6 +2,22 @@
...
+    <meta property="og:url" content="{{ url('author_show', {blogId: blog.getId(), guid: author.getGuid()}) }}" />
+{% endblock %}
+
+{% block social_image_tags %}
+    {%- if author.getImage() -%}
+        <meta itemprop="image" content="{{ author.getImage().getUrl(1200,675) }}">
+        <meta property="og:image" content="{{ author.getImage().getUrl(1200,675) }}" />
+    {%- endif -%}
+
+    {{ parent() }}
```
Comment (votemike) [inline templates/author/show.html.twig]: Do you want the parent? Don't you want to *override* the parent?

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: templates/author/index.html.twig [hunk 8]
```diff
@@ -2,6 +2,11 @@
 
 {% block title %}BBC {{ tr('page_title_blogs') }} - {{ blog.getName() }} - {{ tr('page_title_author_all') }}{% endblock %}
 
+{% block social_tags %}
+    <meta property="og:title" content="{{ block('title') }}">
```
Comment (votemike) [inline templates/author/index.html.twig]: Do you think `{{ parent() }}` would be better than `<meta property="og:title" content="{{ block('title') }}">` ? Or does that take the wrong `block('title')`?

File: templates/base.html.twig [hunk 9]
```diff
@@ -9,15 +9,26 @@
...
+
+    {% block social_type %}
+        <meta property="og:type" content="website">
+    {% endblock %}
+
+    {% block social_image_tags %}
+        <meta property="og:image" content="{{ fallback_social_image.getUrl(1200,675) }}" />
+    {% endblock %}
+
+    {% block social_description %}{% endblock %}
```
Comment (votemike) [inline templates/base.html.twig]: Is this optional or mandatory?

File: templates/author/show.html.twig [hunk 10]
```diff
@@ -2,6 +2,25 @@
...
+{% block social_image_tags %}
+    {%- if author.getImage() -%}
+        <meta property="og:image" content="{{ author.getImage().getUrl(1200,675) }}" />
+    {%- else -%}
+        {{ parent() }}
+    {%- endif -%}
+{% endblock %}
+
+{% block social_description %}
+    {%- if author.getRole() -%}
```
Comment (votemike) [inline templates/author/show.html.twig]: What if the author has a description but not a role?

File: templates/blogs_base.html.twig [hunk 11]
```diff
@@ -25,3 +25,15 @@
...
+{% block social_image_tags %}
+    {%- if blog.getImage() -%}
+        <meta property="og:image" content="{{ blog.getImage().getUrl(1200,675) }}" />
+    {%- else -%}
+        {{- parent() -}}
+    {%- endif -%}
+{% endblock %}
+
+{% block social_description %}
+    <meta property="og:description" content="{{ blog.getDescription() }}" />
```
Comment (votemike) [inline templates/blogs_base.html.twig]: Which pages do not set their own description?

File: templates/post/by_date.html.twig [hunk 12]
```diff
@@ -4,6 +4,11 @@
     BBC {{ tr('page_title_blogs') }} - {{ blog.getName() }} - {{ tr('header_archive_month') }}, {{ datePicker.getChosenMonthYear()|local_date_intl('MMMM') }} {{ datePicker.getChosenMonthYear()|date('Y') -}}
 {%- endblock %}
 
+{% block social_tags %}
+    <meta property="og:title" content="{{ block('title') }}">
+    <meta property="og:url" content="{{ url('posts_year_month', {blogId: blog.getId(), year: datePicker.getChosenMonthYear()|date('Y'), month: datePicker.getChosenMonthYear()|date('m') }) }}" />
```
Comment (votemike) [inline templates/post/by_date.html.twig]: Does this work if there is no month?

**Pull request merged**